heron/common/src/python/utils/log.py

```diff
@@ -26,6 +27,26 @@
 # see time formatter documentation for more
 date_format = "%Y-%m-%d %H:%M:%S %z"
 
+def async_stream_process_stdout(process, log_fn):
+  """ Stream the stdout and stderr for a process out to display async
+  :param process: the process to stream the log for
+  :param log_fn: a function that will be called for each log line
+  :return: None
+  """
+  logging_thread = Thread(target=stream_process_stdout, args=(process, log_fn, ))
+  logging_thread.start()
+
+  return logging_thread
+
+def stream_process_stdout(process, log_fn):
+  """ Stream the stdout and stderr for a process out to display
+  :param process: the process to stream the logs for
+  :param log_fn: a function that will be called for each log line
+  :return: None
+  """
+  for line in iter(process.stdout.readline, b''):
```

Comment:
I use while loop because of [this issue](http://bugs.python.org/issue3907). It seems that Python 2.7 fixed this issue but as you can see in the thread some people are still reporting seeing this issue in 2.7